src/components/Pagination/index.jsx

```diff
@@ -2,35 +2,72 @@ import React from "react";
 import cn from "classnames";
 import "./index.scss";
 
+function lastPageToDisplay(currentPageIndex, pagesToDisplay, totalPages) {
+  const lastPageCalculation =
+    currentPageIndex +
+    (pagesToDisplay -
+      (currentPageIndex % pagesToDisplay === 0
+        ? pagesToDisplay
+        : currentPageIndex % pagesToDisplay));
+  const lastPage =
+    lastPageCalculation > totalPages ? totalPages : lastPageCalculation;
+  return lastPage;
+}
+
+function firstPageToDisplay(endPage, pagesToDisplay) {
+  const startPage = endPage - (pagesToDisplay - 1);
+  return startPage <= 1 ? 1 : startPage;
+}
+
+function calShouldDisplayLastPage(
```

Comment:
let's rename this calc instead since i was like calshouldwhat? haha thanks for abstracting out all the pagination logic into something that can be shared throughout